### PR TITLE
Improve test coverage in the jdk.internal.layout package

### PR DESCRIPTION
**Pull request by minborg**
This PR adds some tests and minor code refactoring.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Reviewers
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/panama-foreign pull/790/head:pull/790` \
`$ git checkout pull/790`

Update a local copy of the PR: \
`$ git checkout pull/790` \
`$ git pull https://git.openjdk.org/panama-foreign pull/790/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 790`

View PR using the GUI difftool: \
`$ git pr show -t 790`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/panama-foreign/pull/790.diff">https://git.openjdk.org/panama-foreign/pull/790.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/foreign/TestLayouts.java [hunk 1]
```diff
@@ -72,6 +76,8 @@ public void testNotEquals(MemoryLayout layout) {
     public void testTargetLayoutEquals() {
         MemoryLayout differentTargetLayout = ADDRESS.withTargetLayout(JAVA_CHAR);
         assertFalse(ADDRESS.equals(differentTargetLayout));
+        var equalButNotSame = ADDRESS.withTargetLayout(JAVA_INT).withTargetLayout(JAVA_CHAR);
+        assertFalse(ADDRESS.equals(equalButNotSame));
```
Comment (JornVernee) [inline test/jdk/java/foreign/TestLayouts.java:80]: I don't see what this extra code is testing that wasn't already tested by the lines above.

Was this intended as `assertTrue(differentTargetLayout .equals(equalButNotSame));` ?

**Review by JornVernee — APPROVED** [review]

File: test/jdk/java/foreign/TestLayouts.java (hunk 1 above)
Comment (minborg) [reply test/jdk/java/foreign/TestLayouts.java:80]: The idea was to create a layout that is equal but not the same so to touch `OfAddress::equals` at the end where the layouts are compared.

Comment (JornVernee) [reply test/jdk/java/foreign/TestLayouts.java:80]: Ok, I get that, but that's not what the test code seems to be doing. i.e.g the 2 layouts are not equal, AFAICS

File: src/java.base/share/classes/java/lang/foreign/PaddingLayout.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- *  Copyright (c) 2019, 2022, Oracle and/or its affiliates. All rights reserved.
+ *  Copyright (c) 2019, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (JornVernee) [inline src/java.base/share/classes/java/lang/foreign/PaddingLayout.java:2]: Seems like a spurious change? No other changes to this file.

**Review by JornVernee — APPROVED** [review]

**Pull request closed without merging**